lib/sshkit/deprecation_logger.rb

```diff
@@ -11,8 +11,14 @@ def log(message)
       return if @out.nil?
       warning_msg = "[Deprecated] #{message}\n"
       caller_line = caller.find { |line| !line.include?('lib/sshkit') }
-      warning_msg << "    (Called from #{caller_line})\n" unless caller_line.nil?
-      @out << warning_msg unless @previous_warnings.include?(warning_msg)
+      warning_msg = "#{warning_msg}    (Called from #{caller_line})\n" unless caller_line.nil?
```

Comment:
I think this change isn't needed because `warning_msg` will not be frozen. This is because `warning_msg` is constructed as in interpolated string (i.e. "...#{...}..."), which are not frozen. I confirmed this in Ruby 3.4.0-preview2:

```
(rbenv local 3.4.0-preview2 && RUBYOPT=--enable=frozen-string-literal irb)
irb(main):001> "hi".frozen?
=> true
irb(main):002> "hi #{'there'}".frozen?
=> false
```